app/services/return-logs/setup/submit-check.service.js

```diff
@@ -37,20 +68,22 @@ async function go(sessionId, user) {
     await CreateReturnLinesService.go(returnSubmission.id, session, timestamp, trx)
 
     await _markReturnLogAsSubmitted(session.returnLogId, session.receivedDate, timestamp, trx)
-    await _cleanupSession(sessionId, trx)
+    await _cleanupSession(session.id, trx)
   })
-
-  return session.returnLogId
 }
 
-async function _markReturnLogAsSubmitted(returnLogId, receivedDate, timestamp, trx) {
-  await ReturnLogModel.query(trx)
-    .patch({ status: 'completed', receivedDate, updatedAt: timestamp })
-    .where({ id: returnLogId })
-}
+function _validate(session) {
+  const validation = CheckValidator.go(session)
 
-async function _cleanupSession(sessionId, trx) {
-  await SessionModel.query(trx).deleteById(sessionId)
+  if (!validation.error) {
+    return null
+  }
+
+  const { message } = validation.error.details[0]
+
+  return {
+    errorList: [{ text: message }]
+  }
```

Comment:
The convention now is to use the new helper.

```suggestion
  return formatValidationResult(validationResult)
```

Reply:
New convention used